### PR TITLE
firefox: update test expectations to transceiver model

### PR DESCRIPTION
**Pull request by fippo**
partial update of the firefox test expectations to better
reflect the FF59+ transceiver model

(still some test failures which might require test updates)

### CODE REVIEW

**Comment by fippo** [comment]
https://travis-ci.org/webrtc/adapter/jobs/334699287#L1827 shows the failures. Some of the tests make an assumption about the number of senders after removeTrack etc.

Not sure what is the best way to deal with this. Mark them as expected failure in Firefox and add new ones which expect transceiver behaviour which are marked as expected failure in Chrome?

@jan-ivar thoughts?

**Comment by jan-ivar** [comment]
I think adapter should follow the spec.

**Comment by KaptenJansson** [comment]
I agree that all the test cases should follow the spec and if they fail for a particular browser and there is no fast way to fix, add it to the expectations as a failure.

In the end, all expectations should be pass/success for all browsers.

File: test/karma.conf.js [hunk 1]
```diff
@@ -73,6 +73,11 @@ module.exports = function(config) {
     preprocessors: {
       'src/js/adapter_core.js': ['browserify']
     },
+    client: {
+      mocha: {
+        timeout: 5000, // 6 second timeout per test
```
Comment (KaptenJansson) [inline test/karma.conf.js]: Code says 5 and comment says 6 seconds?

Comment (KaptenJansson) [reply test/karma.conf.js]: Also did this help?

Comment (fippo) [reply test/karma.conf.js]: not really... will pull out that commit

**Comment by KaptenJansson** [comment]
I guess the addTrack failures should be add as ERR to expectations file, once fixed in Firefox they should fail once again and we can change to PASS. Since the spec spec states InvalidAccessError should be the error name. 

```
FAILED TESTS:
  addTrack
    throws an exception
      ✖ if the track has already been added
        Firefox 58.0.0 (Linux 0.0.0)
      expected 'InvalidParameterError' to equal 'InvalidAccessError'
      AssertionError@node_modules/chai/chai.js:5553:18
      [3]</module.exports/Assertion.prototype.assert@node_modules/chai/chai.js:206:13
      assertEqual@node_modules/chai/chai.js:776:7
      [10]</module.exports/ctx[name]@node_modules/chai/chai.js:4192:18
      @test/e2e/addTrack.js:30:9
      
      ✖ if the track has already been added via addStream
        Firefox 58.0.0 (Linux 0.0.0)
      expected 'InvalidParameterError' to equal 'InvalidAccessError'
      AssertionError@node_modules/chai/chai.js:5553:18
      [3]</module.exports/Assertion.prototype.assert@node_modules/chai/chai.js:206:13
      assertEqual@node_modules/chai/chai.js:776:7
      [10]</module.exports/ctx[name]@node_modules/chai/chai.js:4192:18
      @test/e2e/addTrack.js:42:9
      
      ✖ if addStream is called with a stream containing a track already added
        Firefox 58.0.0 (Linux 0.0.0)
      expected 'InvalidParameterError' to equal 'InvalidAccessError'
      AssertionError@node_modules/chai/chai.js:5553:18
      [3]</module.exports/Assertion.prototype.assert@node_modules/chai/chai.js:206:13
      assertEqual@node_modules/chai/chai.js:776:7
      [10]</module.exports/ctx[name]@node_modules/chai/chai.js:4192:18
      @test/e2e/addTrack.js:55:9
```
      
I guess this is also an issue in Firefox since it does not make sense that you can insert DTMF on a video sender. Test is [here](https://github.com/webrtc/adapter/blob/master/test/e2e/dtmf.js#L30). Requesting a mediaStream with video=true only should not return a mediaStream with an audio track right?

  ```
dtmf
    RTCRtpSender.dtmf
      ✖ does not exist on video senders
        Firefox 58.0.0 (Linux 0.0.0)
      expected { insertDTMF: [Function: insertDTMF],
        ontonechange: [Getter/Setter],
        toneBuffer: [Getter],
        addEventListener: [Function: addEventListener],
        removeEventListener: [Function: removeEventListener],
        dispatchEvent: [Function: dispatchEvent] } to equal null
      AssertionError@node_modules/chai/chai.js:5553:18
      [3]</module.exports/Assertion.prototype.assert@node_modules/chai/chai.js:206:13
      assertEqual@node_modules/chai/chai.js:776:7
      [10]</module.exports/ctx[name]@node_modules/chai/chai.js:4192:18
      @test/e2e/dtmf.js:35:9
```
     
Does firefox support output device enumeration?

  ```
navigator.mediaDevices
    enumerateDevices
      returns
        ✖ some audiooutput devices
          Firefox 58.0.0 (Linux 0.0.0)
        expected undefined to not equal undefined
        AssertionError@node_modules/chai/chai.js:5553:18
        [3]</module.exports/Assertion.prototype.assert@node_modules/chai/chai.js:206:13
        assertEqual@node_modules/chai/chai.js:776:7
        [10]</module.exports/ctx[name]@node_modules/chai/chai.js:4192:18
        @test/e2e/mediaDevices.js:58:13
```
  
Here it looks like firefox returns invalidSender but not as an exception.
      
 ```
 removeTrack
    throws an exception
      ✖ if the sender does not belong to the peerconnection
        Firefox 58.0.0 (Linux 0.0.0)
      expected [Function: invalidSender] to throw an error
      AssertionError@node_modules/chai/chai.js:5553:18
      [3]</module.exports/Assertion.prototype.assert@node_modules/chai/chai.js:206:13
      assertThrows@node_modules/chai/chai.js:1665:5
      [10]</module.exports/ctx[name]@node_modules/chai/chai.js:4192:18
      @test/e2e/removeTrack.js:43:9
```

**Comment by fippo** [comment]
all updated. So far i have left the "remove all tracks, what streams remain" in favor of the current behaviour.  Still deciding if i want to fight @henbos or @jan-ivar on that... ;-)

**Review by KaptenJansson — APPROVED** [review]

**Comment by jan-ivar** [comment]
> expected 'InvalidParameterError' to equal 'InvalidAccessError'

Thanks, I've filed [bug 1437211](https://bugzilla.mozilla.org/show_bug.cgi?id=1437211).

> ✖ does not exist on video senders

Known [bug 1376947](https://bugzilla.mozilla.org/show_bug.cgi?id=1376947).

> Does firefox support output device enumeration?

Working on it. [Bug 934425](https://bugzilla.mozilla.org/show_bug.cgi?id=934425).

> ✖ if the sender does not belong to the peerconnection

Fixed Firefox 59 (with transceivers).

**Pull request merged**